csharp/src/Ice/Reference.cs

```diff
@@ -930,6 +1047,39 @@ void TraceRetry(string message, int attempt, RetryPolicy policy, Exception? exce
                 }
                 proxy.Communicator.Logger.Trace(TraceLevels.RetryCategory, sb.ToString());
             }
+
+            RetryPolicy GetRetryPolicy(
```

Comment:
It looks like 2 local functions merged into one. Why not GetRetryPolicy(reference, response) and another GetRetryPolicy(transportException, sent, isIdempotent)?

Reply:
Right maybe an extension method for extension and an internal one for response

```
RetryPolicy retryPolicy =
    response?.GetRetryPolicy(reference) ??
    exception?.GetRetryPolicy(sent, request.IsIdempotent) ??
    RetryPolicy.NoRetry;
```